src/main/webapp/app/entities/participation/participation.model.ts

```diff
@@ -33,6 +33,7 @@ export abstract class Participation implements BaseEntity {
     public results: Result[];
     public submissions: Submission[];
     public type: ParticipationType;
+    public exercise: Exercise;
```

Comment:
I am not sure if this is a good idea, because the subclasses of Participation in the client already have an attribute `public exercise: Exercise;` or `public programmingExercise: ProgrammingExercise;`

We should at least make sure that `SolutionProgrammingExerciseParticipation` and `TemplateProgrammingExerciseParticipation` don't have the exercise twice then.

Is it possible in TypeScript to refine the type of an attribute in the subclass? Then we could simply rename `public programmingExercise: ProgrammingExercise;` --> `public exercise: ProgrammingExercise;` in `SolutionProgrammingExerciseParticipation` and `TemplateProgrammingExerciseParticipation`.

What do you think?

Reply:
You are right, this would be inconsistent and incorrect for the Template/SolutionParticipation (which I did not test here as they are not relevant for this feature). I removed the attribute and made sure it is not used anymore.
I think the solution we have atm is correct and I didn't think about the change enough.